File: artisanat-backend/src/main/java/com/artisanat_backend/dto/response/ProductResponseDto.java
package com.artisanat_backend.dto.response;

import com.artisanat_backend.enums.Category;
import com.artisanat_backend.enums.Collection;
import com.artisanat_backend.model.Artisan;
import com.artisanat_backend.model.Media;
import com.artisanat_backend.model.Review;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ProductResponseDto {
    private Long id;
    String name;
    String description;
    int rating;
    float price;
    int stock;
    LocalDateTime createdAt = LocalDateTime.now();
    Collection collection;
    Category category;
    Artisan artisan;
    private List<Media> media;
    private List<Review> reviews;
}

File: artisanat-frontend/src/app/core/dtos/response/product-response-dto.ts
import {Collection} from "../../enums/collection.enum";
import {Category} from "../../enums/category.enum";
import {Media} from "../../models/media.model";
import {Review} from "../../models/review.model";
import {Artisan} from "../../models/artisan.model";

export interface ProductResponseDto {
  id: number;
  name: string;
  description: string;
  rating: number;
  price: number;
  stock: number;
  createdAt: string;
  collection: Collection;
  category: Category;
  artisan: Artisan;
  media: Media[];
  reviews: Review[];
}
